astroid/nodes/scoped_nodes/mixin.py

```diff
@@ -50,6 +50,24 @@ def scope(self: _T) -> _T:
         """
         return self
 
+    def scope_lookup(self, node, name: str, offset: int = 0):
+        """Lookup where the given variable is assigned.
+
+        :param node: The node to look for assignments up to.
+            Any assignments after the given node are ignored.
+        :type node: NodeNG
+
+        :param name: The name of the variable to find assignments for.
+
+        :param offset: The line offset to filter statements up to.
+
+        :returns: This scope node and the list of assignments associated to the
+            given name according to the scope where it has been found (locals,
+            globals or builtin).
+        :rtype: tuple(str, list(NodeNG))
+        """
+        raise NotImplementedError
```

Comment:
```suggestion
        raise NotImplementedError  # pragma: no cover
```

Reply:
I'll add this to the last of standard uncovered lines.